components/layout/flex.rs

```diff
@@ -0,0 +1,445 @@
+                if kid_base.flags.contains(INLINE_POSITION_IS_STATIC) {
+                    kid_base.position.start.i =
+                        if kid_mode.is_bidi_ltr() == containing_block_mode.is_bidi_ltr() {
+                            inline_start_content_edge
+                        } else {
+                            // The kid's inline 'start' is at the parent's 'end'
+                            inline_end_content_edge
+                        };
+                }
+                kid_base.block_container_inline_size = content_inline_size;
+                kid_base.block_container_writing_mode = containing_block_mode;
+            }
+        }
+    }
+
+    // TODO(zentner): This function should actually flex elements!
+    // Currently, this is the core of InlineFlow::propagate_assigned_inline_size_to_children() with
+    // fragment logic stripped out.
+    fn inline_mode_assign_inline_sizes(&mut self,
+                                       _layout_context: &LayoutContext,
+                                       inline_start_content_edge: Au,
+                                       _inline_end_content_edge: Au,
+                                       content_inline_size: Au) {
+        let _scope = layout_debug_scope!("flex::inline_mode_assign_inline_sizes");
+        debug!("inline_mode_assign_inline_sizes");
+
+        debug!("content_inline_size = {:?}", content_inline_size);
+        debug!("child_count = {:?}", ImmutableFlowUtils::child_count(self as &Flow) as i32);
+        let even_content_inline_size = content_inline_size / ImmutableFlowUtils::child_count(self as &Flow) as i32;
+
+        let inline_size = self.block_flow.base.block_container_inline_size;
+        let container_mode = self.block_flow.base.block_container_writing_mode;
+        self.block_flow.base.position.size.inline = inline_size;
+
+        let block_container_explicit_block_size = self.block_flow.base.block_container_explicit_block_size;
+        let mut inline_child_start = inline_start_content_edge;
+        for kid in self.block_flow.base.child_iter() {
+            let kid_base = flow::mut_base(kid);
+
+            kid_base.block_container_inline_size = even_content_inline_size;
+            kid_base.block_container_writing_mode = container_mode;
+            kid_base.block_container_explicit_block_size = block_container_explicit_block_size;
+            kid_base.position.start.i = inline_child_start;
+            inline_child_start = inline_child_start + even_content_inline_size;
+        }
+    }
+
+    // TODO(zentner): This function should actually flex elements!
+    fn block_mode_assign_block_size<'a>(&mut self, layout_context: &'a LayoutContext<'a>) {
+        self.block_flow.assign_block_size(layout_context)
+    }
+
+    // TODO(zentner): This function should actually flex elements!
+    // Currently, this is the core of TableRowFlow::assign_block_size() with
+    // float related logic stripped out.
```

Comment:
Surprisingly, yes. However, it's relatively minimal (mostly a call to `kid.place_float_if_applicable()`).